src/clarity-angular/utils/conditional/if-active.directive.ts

```diff
@@ -34,15 +34,22 @@ import {Subscription} from "rxjs/Subscription";
 export class IfActiveDirective implements OnDestroy {
 
     private subscription: Subscription;
+    private wasActive: boolean = false;
 
     constructor( private ifActiveService: IfActiveService, @Inject(IF_ACTIVE_ID) private id: number,
                  private template: TemplateRef<any>, private container: ViewContainerRef ) {
 
         this.updateView(ifActiveService.current === this.id);
 
         this.subscription = this.ifActiveService.currentChange.subscribe((newId) => {
-                this.updateView(newId === this.id);
-                this.activeChange.emit(newId === this.id);
+                let isNowActive = newId === this.id;
+                this.updateView(isNowActive);
+
+                // only emit if the new active state is changed since last time.
+                if (isNowActive !== this.wasActive) {
+                    this.activeChange.emit(isNowActive);
```

Comment:
Shouldn't the `this.updateView(isNowActive);` be inside the `if` ? I'm afraid we might have an edge case in the future where we duplicate a tab content if this observable fires with the same id twice. I don't think it can happen right now, but just putting everything inside the `if` doesn't cost anything and is safer.